src/lib.rs

```diff
@@ -89,6 +89,8 @@ pub enum Exception {
     Read(PathBuf),
 
     /// Allow write access to the path and anything beneath it.
+    ///
+    /// On Linux this will **always** also grant read access to the path.
     Write(PathBuf),
```

Comment:
In theory we could change this to `WriteAndRead` similar to `ExecuteAndRead` and automatically also add read on macOS to make things consistent.

That might be better, but it would make it a breaking change if we ever change it back to allow writing without read access. This could be accomplished by layering landlock on top of the namespaces for example.

Reply:
+1 for the suggestion to change to `WriteAndRead`. There are bound to be more breaking changes as this "zero-ver" library evolves and should not cause heartache to make things better along the way.